LiteCore/Support/Error.cc

```diff
@@ -379,14 +450,58 @@ namespace litecore {
     }
 #endif // LITECORE_IMPL
 
+    error error::make_error(Domain domain, int code) {
+        if(domain == POSIX && (code <= MinPosixErrorMinus1 || code >= MaxPosixErrorPlus1)) {
+            for (const codeMapping *row = &kPOSIXMapping[0]; row->err != 0; ++row) {
+                if(row->err == code) {
+                    return error(row->domain, row->code, _what(row->domain, row->code));
+                }
+            }
+
+            DebugAssert(false, "Invalid POSIX code for make_error: %d", code);
+            return error(error::UnexpectedError);
+        }
+
+        return error(domain, code, _what(domain, code));
+    }
+
+
+    error error::make_error(Domain domain, int code, const string& what) {
+        if(domain == POSIX && (code <= MinPosixErrorMinus1 || code >= MaxPosixErrorPlus1)) {
```

Comment:
intentional?

```suggestion
        if(domain == POSIX && (code < MinPosixErrorMinus1 || code > MaxPosixErrorPlus1)) {
```

Reply:
MinPosixErrorMinus1 is one before the first valid error code, and MaxPosixErrorPlus1 is one after, as indicated by their names.  Neither one is a valid error code.

Reply:
right